kernel: kmod-fs-xfs: make it compatible with V4

danitool opened the pull request:
Since kernel commit [b96cb835e37c2ca03aaceef9555ec9047a422d91](https://github.com/torvalds/linux/commit/b96cb835e37c2ca03aaceef9555ec9047a422d91), support
for V4 needs to explicitly enabled. This change won't increase the
firmware size.

Add support for deprecated V4 XFS which will be supported until 2030.

danitool committed:
kernel: kmod-fs-xfs: make it compatible with V4

Since kernel commit b96cb835e37c2ca03aaceef9555ec9047a422d91, support
for V4 needs to explicitly enabled. This change won't increase the
firmware size.

Add support for deprecated V4 XFS which will be supported until 2030.

Signed-off-by: Daniel González Cabanelas <dgcbueu@gmail.com>

Ansuel commented:
@danitool reading the commit, V4 does more harm than good so it's not a bad idea to deprecate. Does not enabling it cause problem/it doesn't work at all? Or it's just adding support for v4?

danitool commented:
@Ansuel, it's for keeping V4 support available. In some devices like NAS, if OpenWrt is installed for replacing the OEM firmware, which previously was managing a hard drive with partitions formated in XFS (very common), it won't recognize those partitions because they were using the V4 format. 

So this is basically to keep backward compatibility with stock firmware. E.g: Buffallo Linkstations NASes

danitool committed:
ipq40xx: Linskys MR8300: fix the USB power port

The USB port on the MR8300 randomly fails to feed bus-powered devices.

This is caused by a misconfigured pinmux. The GPIO68 is used to enable
the USB power (active low), but it's inside the NAND pinmux.

This GPIO pin was found in the original firmware at a startup script in
both MR8300 and EA8300. Therefore apply the fix for both boards.

Signed-off-by: Daniel González Cabanelas <dgcbueu@gmail.com>

danitool committed:
Merge branch 'openwrt:master' into master

danitool committed:
undo
